### PR TITLE
mgmg/mcumgr/lib: unify shell management code

### PR DESCRIPTION
**Pull request by de-nordic**
The PR contains two commits:
1) removes Zephyr specific source file and puts its contents into common source file;
2) removes command line initialization to all 0 as it is not needed.

### CODE REVIEW

File: subsys/mgmt/mcumgr/lib/cmd/shell_mgmt/src/shell_mgmt.c
Comment (lairdjm) [inline subsys/mgmt/mcumgr/lib/cmd/shell_mgmt/src/shell_mgmt.c:43]: will this being static have issues if 2 SMP interfaces are active and both send a shell exec function at the same or similar time?

Comment (de-nordic) [reply subsys/mgmt/mcumgr/lib/cmd/shell_mgmt/src/shell_mgmt.c:43]: Yes it will, you are right. I have failed to notice "static".
Removal of the static will increase required stack size. I should probably do that and update Kconfig for shell management to notify users on possible increased stack requirements.

**Review by carlescufi — DISMISSED** [review]

**Review by nvlsianpu — DISMISSED** [review]

**Review by nordicjm — DISMISSED** [review]

**Review by carlescufi — DISMISSED** [review]

**Review by nvlsianpu — DISMISSED** [review]

File: subsys/mgmt/mcumgr/lib/cmd/shell_mgmt/src/shell_mgmt.c
Comment (lairdjm) [inline subsys/mgmt/mcumgr/lib/cmd/shell_mgmt/src/shell_mgmt.c:15]: Duplicate include

Comment (nvlsianpu) [reply subsys/mgmt/mcumgr/lib/cmd/shell_mgmt/src/shell_mgmt.c:15]: It is because this must be included for sure, we need the redundancy in case the firs line doesn't work.

Comment (nvlsianpu) [reply subsys/mgmt/mcumgr/lib/cmd/shell_mgmt/src/shell_mgmt.c:15]: just kidding

Comment (nvlsianpu) [reply subsys/mgmt/mcumgr/lib/cmd/shell_mgmt/src/shell_mgmt.c:15]: again, @de-nordic must fix this using a rebase, which will dismiss all approvals, and then he must wait on approvals again,

Comment (carlescufi) [reply subsys/mgmt/mcumgr/lib/cmd/shell_mgmt/src/shell_mgmt.c:15]: So be it, we can't merge this with duplicate includes. @de-nordic please fix it and we'll re-approve and merge

**Review by nordicjm — APPROVED** [review]

File: subsys/mgmt/mcumgr/lib/cmd/shell_mgmt/src/shell_mgmt.c
Comment (de-nordic) [reply subsys/mgmt/mcumgr/lib/cmd/shell_mgmt/src/shell_mgmt.c]: > It is because this must be included for sure, we need the redundancy in case the first line doesn't work.

Never trust the preprocessor.

**Review by carlescufi — APPROVED** [review]

**Pull request merged**
